### ISSUE TITLE
cannot restore full data  

### ISSUE THREAD

**Issue by SimonWXH** [iSoron/uhabits#822]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] I am running the latest version of Loop Habit Tracker.
- [x] I have have searched for similar issues, but did not find any matches.

## Description
After tap "Import data"from settings ,the app crashed. No file browser appeared.

## System information
 - Phone: Redmi K30 PRO
 - Phone Operating System: Miui 12.2.3 base on android 11
 - App version: 1.8.12 from f-droid

**Comment by hiqua** [comment]
Can't reproduce on an Android 11 emulator. Just in case, does it work better with the v2-alpha? https://github.com/iSoron/uhabits/releases/tag/v2.0.0-alpha

**Comment by iSoron** [comment]
@SimonWXH Do you have a file browser installed in your phone? Loop 1.8 will crash if no file browser is installed (#181).

**Comment by SimonWXH** [comment]
@hiqua  @iSoron  Very thanks for your help, I have installed solid explorer from google play, here is the error report generated by uhabits:

```---------- BUG REPORT BEGINS ----------
App Version Name: 1.8.12
App Version Code: 10812
OS Version: 4.19.113-perf-gee8793f (V12.2.3.0.RJKCNXM)
OS API Level: 30
Device: lmi
Model (Product): Redmi K30 Pro (lmi)
Manufacturer: Xiaomi
Other tags: release-keys
Screen Width: 1080
Screen Height: 2270
External storage state: mounted


--------- beginning of system
03-07 21:28:26.484 27350 27350 I Timeline: Timeline: Activity_launch_request time:62669485
03-07 21:28:33.751 27350 27350 I Timeline: Timeline: Activity_launch_request time:62676752
03-07 21:28:35.382 27350 27350 I Timeline: Timeline: Activity_launch_request time:62678383
--------- beginning of crash
03-07 21:28:35.385 27350 27350 E AndroidRuntime: FATAL EXCEPTION: main
03-07 21:28:35.385 27350 27350 E AndroidRuntime: Process: org.isoron.uhabits, PID: 27350
03-07 21:28:35.385 27350 27350 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=7, result=1, data=null} to activity {org.isoron.uhabits/org.isoron.uhabits.MainActivity}: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.OPEN_DOCUMENT cat=[android.intent.category.OPENABLE] typ=*/* }
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5108)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5149)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2108)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:236)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7876)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: Caused by: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.OPEN_DOCUMENT cat=[android.intent.category.OPENABLE] typ=*/* }
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2075)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1731)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.Activity.startActivityForResult(Activity.java:5389)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:767)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.Activity.startActivityForResult(Activity.java:5347)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:754)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at org.isoron.uhabits.activities.habits.list.ListHabitsScreen.showImportScreen(ListHabitsScreen.kt:171)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at org.isoron.uhabits.activities.habits.list.ListHabitsScreen.onSettingsResult(ListHabitsScreen.kt:122)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at org.isoron.uhabits.activities.habits.list.ListHabitsScreen.onResult(ListHabitsScreen.kt:101)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at org.isoron.androidbase.activities.BaseActivity.onActivityResult(BaseActivity.java:108)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:8460)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5101)
03-07 21:28:35.385 27350 27350 E AndroidRuntime: 	... 11 more
03-07 21:28:35.387 27350 27350 D OOMEventManagerFK: checkEventAndDumpForJE: 0
--------- beginning of main
03-08 08:23:46.715 27646 27676 D ViewContentFactory: initViewContentFetcherClass
03-08 08:23:46.715 27646 27676 I ContentCatcher: ViewContentFetcher : ViewContentFetcher
03-08 08:23:46.715 27646 27676 D ViewContentFactory: createInterceptor took 0ms
03-08 08:23:46.720 27646 27676 I ContentCatcher: Interceptor : Catcher list invalid for org.isoron.uhabits@org.isoron.uhabits.MainActivity@153055697
03-08 08:23:46.720 27646 27676 I ContentCatcher: Interceptor : Get featureInfo from config image_pick_mode
03-08 08:23:46.720 27646 27676 I ContentCatcher: Interceptor : Get featureInfo from config pick_mode
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: QUALCOMM build                   : 191610ae03, Ic907de5ed0
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: Build Date                       : 09/17/20
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.32.02.01
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: Local Branch                     : 
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: Remote Branch                    : refs/tags/AU_LINUX_ANDROID_LA.UM.9.12.10.00.00.582.274
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: Remote Branch                    : NONE
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: Reconstruct Branch               : NOTHING
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: Build Config                     : S P 10.0.5 AArch64
03-08 08:23:46.749 27646 27673 I AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
03-08 08:23:46.754 27646 27673 I AdrenoGLES-0: PFP: 0x016dd091, ME: 0x00000000
03-08 08:23:48.841 27646 27658 W .isoron.uhabit: Reducing the number of considered missed Gc histogram windows from 171 to 100
03-08 08:23:49.999 27646 27646 I Timeline: Timeline: Activity_launch_request time:64406911
03-08 08:23:50.029 27646 27676 D ViewContentFactory: initViewContentFetcherClass
03-08 08:23:50.029 27646 27676 I ContentCatcher: ViewContentFetcher : ViewContentFetcher
03-08 08:23:50.029 27646 27676 D ViewContentFactory: createInterceptor took 0ms
03-08 08:23:50.030 27646 27676 I ContentCatcher: Interceptor : Catcher list invalid for org.isoron.uhabits@org.isoron.uhabits.activities.settings.SettingsActivity@238361867
03-08 08:23:50.030 27646 27676 I ContentCatcher: Interceptor : Get featureInfo from config image_pick_mode
03-08 08:23:50.030 27646 27676 I ContentCatcher: Interceptor : Get featureInfo from config pick_mode
03-08 08:23:50.332 27646 27673 D OpenGLRenderer: endAllActiveAnimators on 0x7804b0b100 (MenuPopupWindow$MenuDropDownListView) with handle 0x785e7430c0
---------- BUG REPORT ENDS ------------
```

**Comment by iSoron** [comment]
Thank you for the log file, @SimonWXH. This confirms that the problem is that your phone does not have a file browser that can handle this action. I would expect Solid Explorer to handle it, but apparently it doesn't. Unfortunately, there is nothing we can do in this case. I suggest trying another file browser. Newer versions of the app will make this error message more clear (instead of just crashing).

In very early version of the app, we used to implement our own (simplified) file browser, but we have since switched to using Storage Access Framework, following Google's recommendations.
